packages/myst-cli/src/build/typst/single.ts

```diff
@@ -0,0 +1,303 @@
+  const tex = pipe.stringify(result);
+  logMessagesFromVFile(session, tex);
+  return tex.result as TypstResult;
+}
+
+export function extractTexPart(
+  session: ISession,
+  mdast: Root,
+  references: References,
+  partDefinition: TemplatePartDefinition,
+  frontmatter: PageFrontmatter,
+  templateYml: TemplateYml,
+): TypstResult | undefined {
+  const part = extractPart(mdast, partDefinition.id);
+  if (!part) return undefined;
+  const partContent = mdastToTypst(session, part, references, frontmatter, templateYml);
+  return partContent;
+}
+
+export async function localArticleToTexRaw(
+  session: ISession,
+  templateOptions: ExportWithOutput,
+  projectPath?: string,
+  extraLinkTransformers?: LinkTransformer[],
+): Promise<ExportResults> {
+  const { article, output } = templateOptions;
+  const [{ mdast, frontmatter, references }] = await getFileContent(
+    session,
+    [article],
+    path.join(path.dirname(output), 'files'),
+    {
+      projectPath,
+      imageAltOutputFolder: 'files/',
+      imageExtensions: TYPST_IMAGE_EXTENSIONS,
+      extraLinkTransformers,
+      simplifyOutputs: true,
+    },
+  );
+  const toc = tic();
+  const result = mdastToTypst(session, mdast, references, frontmatter, null);
+  session.log.info(toc(`📑 Exported TeX in %s, copying to ${output}`));
+  // TODO: add imports and macros?
+  writeFileToFolder(output, result.value);
+  return { tempFolders: [] };
+}
+
+function writeBibtexFromCitationRenderers(session: ISession, output: string) {
+  const cache = castSession(session);
+  const allBibtexContent = Object.values(cache.$citationRenderers)
+    .map((renderers) => {
+      return Object.values(renderers).map((renderer) => {
+        const bibtexContent = (renderer.cite._graph as any[]).find((item) => {
+          return item.type === '@biblatex/text';
+        });
+        return bibtexContent?.data;
+      });
+    })
+    .flat()
+    .filter((item) => !!item);
+  const bibtexContent = [...new Set(allBibtexContent)].join('\n');
+  if (!fs.existsSync(output)) fs.mkdirSync(path.dirname(output), { recursive: true });
+  fs.writeFileSync(output, bibtexContent);
+}
+
+export async function localArticleToTexTemplated(
+  session: ISession,
+  file: string,
+  templateOptions: ExportWithOutput,
+  projectPath?: string,
+  force?: boolean,
+  extraLinkTransformers?: LinkTransformer[],
+): Promise<ExportResults> {
+  const filesPath = path.join(path.dirname(templateOptions.output), 'files');
+  const [{ frontmatter, mdast, references }] = await getFileContent(
+    session,
+    [templateOptions.article],
+    filesPath,
+    {
+      projectPath,
+      imageAltOutputFolder: 'files/',
+      imageExtensions: TYPST_IMAGE_EXTENSIONS,
+      extraLinkTransformers,
+      simplifyOutputs: true,
+    },
+  );
+  writeBibtexFromCitationRenderers(
+    session,
+    path.join(path.dirname(templateOptions.output), DEFAULT_BIB_FILENAME),
+  );
+
+  const mystTemplate = new MystTemplate(session, {
```

Comment:
This template stuff will be easier once there are actually `typst` templates. Right now there's a bunch of copied code that doesn't actually do anything I don't think.